changelogs/fragments/111_add_weos_network_cli_plugins.yml

```
@@ -0,0 +1,4 @@
+---
+minor_changes:
```

```suggestion
minor_changes:
```
no changelog needed for new plugins / modules

also all new stuff must have tests (integration and / or unit)

---

Have been busy om some other projects, just commited the unit tests and renamed the module to WeOS4 since WeOS5 is very different and will need it's own module.

---

can't we detect which version of WeOS is used and define and run corresponding behavior dynamically?

---

We could probably do that, but since the cli and configuration format (json style instead of indented). Most of the cliconf functions would be completely different for v4 and v5. It would also be slower since for every operation the version would have to be read from the device.

---

if it's possible, i'd suggest implementing a generalized solution even if it's slower (can't we write, say, two classes for 4 and 5 / or add an attribute containing the version, then detect the version once, and initialize a corresponding class / run corresponding code inside the methods?).

---

That could probably be done, but I don't see any other cliconf plugins doing something like that. Also WeOS 4 and WeOS 5 run on different hardware platforms. It's a similar situation to the DellOS6/9/10 modules.

---

Ok, thanks for the explanation